istio/src/main/scala/io/buoyant/k8s/istio/CurrentIstioPath.scala

```diff
@@ -6,26 +6,65 @@ import com.twitter.logging.Logger
 import io.buoyant.router.context.DstBoundCtx
 
 object CurrentIstioPath {
-  val log = Logger(this.getClass.getName)
-  private val pathLength = 10
+  val logger = Logger(this.getClass.getName)
 
   def apply(maybeBound: Option[Dst.Bound]): Option[Path] = {
     maybeBound.flatMap { bound =>
       bound.id match {
-        case path: Path if (path.elems.length == pathLength) => Some(path)
-        case _ => None
+        case path: Path => Some(path)
+        case other =>
+          logger.warning("Expected path, got %s", other)
+          None
       }
     }
   }
 
+  def targetServiceIn(istioPath: Path): Option[String] = {
```

Comment:
It would be helpful to have a comment with the expected structure of path so that these offsets are easier to understand.

Reply:
Will do